demo/angular-editable-text.js

```diff
@@ -42,49 +43,80 @@
 
           elem.addClass('gg-editable-text');
 
-          scope.$watch('isEditing', function (val, oldVal) {
+          scope.onInputClick = function() {
+            scope.isEditing = true;
+            if (attrs.hasOwnProperty('selectAll')) {
+              input.select();
+            }
+          };
+
+          $timeout(function() {
+            $(input[0]).width($(elem).width());
+          });
+
+          scope.onInputBlur = function() {
+            scope.isEditing = false;
+            // Kind of a hacky way, would be great to not have to do this
+            $timeout(function() {
+              $(input[0]).width($(elem).width());
+            });
+          };
+
+          scope.onKeyPress = function(e) {
+            var inputElem = input[0];
+            if (e.which === 13) {
+              $(inputElem).blur();
+            } else if (e.which === 27) {
+              scope.editingValue = scope.editableText;
+              $(inputElem).blur();
+            }
+          };
+
+          scope.$watch('isEditing', function(val, oldVal) {
```

Comment:
I think it'd be easier to follow what is going on if `val` was renamed to `isEditing`.  Same down in the other watch.